### DISCUSSION TITLE
Would be awe for Async.parallel to not allocate array on Async<unit>

### DISCUSSION THREAD

**Discussion by Xyncgas** [dotnet/fsharp#16183] [General]
Async.Parallel returns array even the input was Async<unit> seq
It would be better to execute these Async operations and waiting on all to finish without allocating an array
Is converting async to .NET TPL in Async.Parallel(input :Async<unit> seq) overload and using Task.Whenall an acceptable solution?

**Comment by T-Gro** [comment]
You mean that FSharp.Core librabry would without user's explicit intention convert from async to task and then back?
It's fine for an opinionated library/wrapper served from a 3rd party, but should not be present in the standard library.
